pkg/cmd/server/start/master_args.go

```diff
@@ -254,6 +254,7 @@ func (args MasterArgs) BuildSerializeableMasterConfig() (*configapi.MasterConfig
 		if builtInKubernetes {
 			config.KubeletClientInfo.CA = admin.DefaultRootCAFile(args.ConfigDir.Value())
 			config.KubeletClientInfo.ClientCert = kubeletClientInfo.CertLocation
+			config.ServiceAccountConfig.MasterCA = admin.DefaultRootCAFile(args.ConfigDir.Value())
```

Comment:
> feels like this should move down to the section where we set config.ServiceAccountConfig.PublicKeyFiles... token auth is still applicable even if we're not serving using TLS

What does this have to do with token auth?  It's about confirming that you're actually talking to the master.  In fact, it only exists if TLS is enabled, right?